cli/templatecreate.go

```diff
@@ -41,6 +44,30 @@ func (r *RootCmd) templateCreate() *clibase.Cmd {
 			r.InitClient(client),
 		),
 		Handler: func(inv *clibase.Invocation) error {
+			if failureTTL != 0 || inactivityTTL != 0 {
+				// This call can be removed when workspace_actions is no longer experimental
+				experiments, exErr := client.Experiments(inv.Context())
+				if exErr != nil {
+					return xerrors.Errorf("get experiments: %w", exErr)
+				}
+
+				if !experiments.Enabled(codersdk.ExperimentWorkspaceActions) {
+					return xerrors.Errorf("--failure-ttl and --inactivityTTL are experimental features. Use the workspace_actions CODER_EXPERIMENTS flag to set these configuration values.")
+				}
+
+				entitlements, err := client.Entitlements(inv.Context())
+				var sdkErr *codersdk.Error
+				if xerrors.As(err, &sdkErr) && sdkErr.StatusCode() == http.StatusNotFound {
+					return xerrors.Errorf("your deployment appears to be an AGPL deployment, so you cannot set --failure-ttl or --inactivityTTL")
+				} else if err != nil {
+					return xerrors.Errorf("get entitlements: %w", err)
+				}
+
+				if !entitlements.Features[codersdk.FeatureAdvancedTemplateScheduling].Enabled {
+					return xerrors.Errorf("your license is not entitled to use advanced template scheduling, so you cannot set --failure-ttl or --inactivityTTL")
+				}
```

Comment:
This is nice. For anyone else, the backend allows these inputs and just ignores them for AGPL, so it's good to tell the user up front.

Can we factor this into a function in this file? Since both commands check the same entitlement?

Reply:
Definitely!

Reply:
@Emyrk I looked into this refactor today and idk if it's worth it rn - both commands use the same entitlement, but they have different args, logic branches, and messaging. We'd also have to pass forward both the context and the client. 
Seems like maybe we defer to the [rule of 3](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) here and refactor if this is re-used again?